non functional change to force CI

egstern opened the pull request:
Non functional change to force CI to run. Check whether the macos-pixi CI has rotted.

egstern committed:
Added Lattice_element methods
    .remove_double_attribute(name)
    .remove_string_attribute(name)
    .remove_vector_attribute(name)

to remove the attributes from the lattice element and a
added test_lattice_element.

egstern committed:
non functional change to force CI

egstern committed:
non functional change to force CI

egstern committed:
additional change

egstern committed:
additional change

egstern committed:
Use newly added remove_<double/string>_attribute functionality to
implement a sanitize argument on Lattice::export_madx_file() method that
eliminates MadX non-compliant attributes for elements on output.

egstern commented:
CI:macos-pixi-conda has spontaneously started failing for the unaltered devel3 branch. I can't maintain this so I am disabling that CI.

egstern committed:
add forgotten attribute kl

egstern committed:
add setprecision() calls for better fitting together of beamline elements

egstern committed:
increase precision on sequence length and position of elements

egstern committed:
Merge pull request #342 from fnalacceleratormodeling/egstern/341-remove-lattice-element-attributes

Added Lattice_element methods

s-sajid-ali commented:
Potentially related to something I came across recently: https://lucascolley.github.io/blog/2025-09-26-macos-26-c-compiler/

egstern committed:
checkout kokkos version 4.7.01 to update submodule

egstern committed:
Merge pull request #344 from fnalacceleratormodeling/update-to-kokkos-4.7

checkout kokkos version 4.7.01 to update submodule

egstern committed:
Allow beam_plot.py to work on legacy (Synergia 2) files that don't contain
the particles_masks object. If particles_masks is not present, the beam_plot.py
script will create an array of all true of the correct size.

egstern committed:
Merge pull request #345 from fnalacceleratormodeling/egstern/enable-beamplot-legacy-files

Allow beam_plot.py to work on legacy (Synergia 2) files that don't co…

egstern committed:
new normal form examples detailed in README.md

egstern committed:
Merge pull request #348 from fnalacceleratormodeling/egstern-347-nonlinear-propagation-and-normal-form-example

new normal form examples detailed in README.md

egstern committed:
updates to read openPMD diagnostic outputs

egstern committed:
Merge pull request #349 from fnalacceleratormodeling/egstern/update-openpmd-diagnostic-scripts

updates to read openPMD diagnostic outputs

egstern committed:
update license for new contractor

egstern committed:
Merge pull request #351 from fnalacceleratormodeling/350-update-license-for-new-doe-contract

update license for new contractor

egstern committed:
python wrap and test get_*_names() methods in MadX_reader

egstern committed:
try to install open-mpi in brew for macos CI

egstern committed:
Merge pull request #354 from fnalacceleratormodeling/353-madx_reader-missing-get-line-names-methods

python wrap and test get_*_names() methods in MadX_reader

Also the macos-clang CI stopped working. Github Copilot indicates because it couldn't find MPI.

Problem: The macOS-clang CI job is failing because CMake cannot find MPI (both C and C++ components). The error occurs at line 191 in CMakeLists.txt:

 I added open-mpi to the brew build instructions.

egstern committed:
non functional change

egstern committed:
fix merge conflict
